### DISCUSSION TITLE
Fork/Join vs UserTask

### DISCUSSION THREAD

**Discussion by bryanvam** [elsa-workflows/elsa-core#2516] [Q&A]
I need to do a multi-user document approval with approve and reject. If its rejected at any point the workflow should finish. Going through the documentation I can see that Fork/Join with signals and UserTask are very similar. What what be the best activity to use in a case like this? I currently am using Fork/Join but I am trying to learn this library as I need to use it in other projects as well.

**Comment by sfmskywalker** [comment]
Probably the **Fork**/**Join** activities combined with the **Signal Received** activity are most useful. However, it usually depends on the little details.
